### PR TITLE
ekf2-mag_auto: always use mag 3D after takeoff

### PR DESCRIPTION
**Pull request by bresch**
### Solved Problem
The mag heading fusion might be more robust in some cases, but doesn't consider the mag bias uncertainty and is affecting the heading quite strongly (because it is a direct heading observation, without bias). To prevent this, we're currently allowing it to fuse a mag sample only if the heading uncertainty is becoming too large. 
This protection has other sides effects as it under-constrains the heading when the mag bias have been correctly estimated. This degrades dead-reckoning performance and creates large position corrections every time a new mag sample is fused.

### Solution
Mag 3D fusion is the ideal way of fusing magnetometer measurements in an EKF as the original measurement is fused (instead of preprocessing it using the states -> correlation with the filter). Since the mag bias uncertainty is initially large, the heading isn't over-constrained at takeoff and an incorrect initial heading can easily be corrected through GNSS fusion or angular motions.

As a first step, this PR changes the mag "auto" logic to switch into mag 3D fusion right after the "in flight mag reset" occurring at 1.5m after takeoff. If no issue is observed, I would like to completely remove the mag heading fusion feature as this would greatly simplify the code and remove complicated code paths.

### Changelog Entry
For release notes:
```
Motion is no longer required to start mag 3D fusion; it will now automatically start after the in-air mag reset and stay ON.
New parameter: removed `EKF2_MAG_YAWLIM`
Documentation: No
```

### Alternatives
I tried to compute the actual mag heading uncertainty using the quaternion and mag bias covariance, but the result is still under-estimated.

### Test coverage
SITL tests:
- Initial heading error of 90 degrees due to wrong mag bias calibration (0.2Gs on x and y). Heading and mag biases are quickly corrected.
Screenshot from 2024-02-14 16-57-38

- Incorrect mag data plumbing (mag_x = -mag_y, mag_z = mag_z - mag_x); tested with and without GNSS fusion. In this case, the mag fusion is easily rejected; the attitude is not affected and the heading is correct when fusion is active.

### CODE REVIEW

File: src/modules/ekf2/EKF/mag_3d_control.cpp [hunk 1]
```diff
@@ -65,8 +64,6 @@ void Ekf::controlMag3DFusion(const magSample &mag_sample, const bool common_star
 				       && _control_status.flags.mag_aligned_in_flight
```
Comment (dagar) [inline src/modules/ekf2/EKF/mag_3d_control.cpp:64]: TODO (later): Could we do something to safely allow full mag_3D before mag_aligned_in_flight (eg ground testing, etc)?

**Comment by dagar** [comment]
Update EKF2_MAG_ACCLIM description? 

https://github.com/PX4/PX4-Autopilot/blob/c697c0397e4cfc8c0b112b10bc6e3ee3416a6039/src/modules/ekf2/ekf2_params.c#L505-L516

File: src/modules/ekf2/EKF/mag_3d_control.cpp (hunk 1 above)
Comment (bresch) [reply src/modules/ekf2/EKF/mag_3d_control.cpp:64]: Yes, good point. If this work we could get rid of the separate mag bias estimator used for quick calibration and all the complicated logic that merges both results.

**Comment by RomanBapst** [comment]
@bresch I tested this PR with the deadreckoning scenario and it seems to work fine, I was not able to introduce the position jumps anymore, and in general the dead-reckoning performance is much better.

**Comment by bresch** [comment]
> I tested this PR with the deadreckoning scenario and it seems to work fine, I was not able to introduce the position jumps anymore, and in general the dead-reckoning performance is much better.

Great, thanks. I just need to investigate the optical flow unit test failure and we should be good to go.

File: src/modules/ekf2/EKF/mag_3d_control.cpp [hunk 2]
```diff
@@ -45,13 +45,12 @@ void Ekf::controlMag3DFusion(const magSample &mag_sample, const bool common_star
 
 	resetEstimatorAidStatus(aid_src);
 
-	const bool wmm_updated = (_wmm_gps_time_last_set > aid_src.time_last_fuse);
+	const bool wmm_updated = (_wmm_gps_time_last_set >= aid_src.time_last_fuse); // WMM update can occur on the last epoch, just after mag fusion
```
Comment (dagar) [inline src/modules/ekf2/EKF/mag_3d_control.cpp:48]: I'm wondering if we even need this given the updated WMM is immediately used in by limitDeclination in fuseMag(). Something to review later.

**Pull request merged**